File: backend/app/api/routers/schedules.py
"""Schedules — recurring, unattended workflow triggers. Thin controllers over
ScheduleService. A scheduled run still pauses at its approval gate."""
from __future__ import annotations

from fastapi import APIRouter, Depends

from ...services.scheduling import ScheduleService
from ..deps import User, current_user, get_schedule_service
from ..schemas import ScheduleBody, ToggleBody

router = APIRouter(prefix="/api", tags=["schedules"])


@router.get("/schedules")
def list_schedules(user: User = Depends(current_user),
                   svc: ScheduleService = Depends(get_schedule_service)):
    return svc.list(user.org_id)


@router.post("/schedules")
def create_schedule(body: ScheduleBody, user: User = Depends(current_user),
                    svc: ScheduleService = Depends(get_schedule_service)):
    return svc.create(user.org_id, body.workflow_id, body.params,
                      body.interval_minutes)


@router.post("/schedules/{sched_id}/toggle")
def toggle_schedule(sched_id: str, body: ToggleBody,
                    user: User = Depends(current_user),
                    svc: ScheduleService = Depends(get_schedule_service)):
    svc.set_enabled(sched_id, user.org_id, body.enabled)
    return {"ok": True}


@router.delete("/schedules/{sched_id}", status_code=204)
def delete_schedule(sched_id: str, user: User = Depends(current_user),
                    svc: ScheduleService = Depends(get_schedule_service)):
    svc.delete(sched_id, user.org_id)

File: frontend/src/lib/api/resources/schedules.ts
import { req } from "../http";

export interface Schedule {
  id: string;
  workflow_id: string;
  params: Record<string, string>;
  interval_minutes: number;
  enabled: boolean;
  created_at: string;
  last_run_at: string | null;
  next_run_at: string;
}

export const schedulesApi = {
  listSchedules: () => req<Schedule[]>("/api/schedules"),
  createSchedule: (workflow_id: string, params: Record<string, string>,
                   interval_minutes: number) =>
    req<Schedule>("/api/schedules", {
      method: "POST",
      body: JSON.stringify({ workflow_id, params, interval_minutes }),
    }),
  toggleSchedule: (id: string, enabled: boolean) =>
    req<{ ok: boolean }>(`/api/schedules/${id}/toggle`, {
      method: "POST", body: JSON.stringify({ enabled }),
    }),
  deleteSchedule: (id: string) =>
    req<void>(`/api/schedules/${id}`, { method: "DELETE" }),
};
